### ISSUE TITLE
'skip' argument results in Error: The size of the connection buffer (131072) was not large enough

### ISSUE THREAD

**Issue by mgei** [r-lib/vroom#361]
I have already [asked on Stackoverflow](https://stackoverflow.com/questions/68628271/partially-read-really-large-csv-gz-in-r-using-vroom).

I have a very large .csv.gz file. I can read the first couple of million lines by setting the 'n_max' argument (with vroom this works in no time, wow!).

Now I would like to read further lines, skipping the first couple of million lines.

```
library(vroom)
df <- vroom("HUGE.csv.gz", delim= ",", n_max = 10^7, skip = 10^7)
``` 

This results in the error

```
Error: The size of the connection buffer (131072) was not large enough
to fit a complete line:
  * Increase it by setting `Sys.setenv("VROOM_CONNECTION_SIZE")`
```

I tried to increase with `Sys.setenv("VROOM_CONNECTION_SIZE" = 131072*1000)`, however, the error persists.

**Comment by jimhester** [comment]
I think the problem is that `n_max` and `skip` cannot be the same value, you would need to increase n_max higher if you want to read the second million rows.

**Comment by mgei** [comment]
The problem is likely that random access is no possible on gz compressed files (https://stackoverflow.com/questions/14225751/random-access-to-gzipped-files). The lines have to be read from top to bottom, so even if I were able to set a `skip`, it would just skip the rows after they have been read. This is inefficient and would not increase speed.

My solution was to first decompress and save. Skip worked on the simple `csv`.

**Comment by skchronicles** [comment]
#### Background
I ran into a similar issue as mentioned above when running the CRAN package sequenza.

I had a script that was failing immediately after calling `sequenza.extract()` with a gzipped input file:
```
Error: The size of the connection buffer (131072) was not large enough
to fit a complete line:
  * Increase it by setting `Sys.setenv("VROOM_CONNECTION_SIZE")`
```

I tested it out on a cluster with multiple versions of R installed, each installation has sequenza and its dependencies installed. 

#### Solution
When testing it with R/3.5, everything works fine and as expected; however, **when running the same script with R/4.0, it fails immediately!** I tried different things to resolve the issue, and I discovered that when I provide a gzipped input file as input with R/4.0, it fails. If I uncompress the file first and then run the same script with the uncompressed input file, it works fine.

#### TLDR
I am not sure what has changed across each major version of R but if you are using R/4.0, I would suggest using an un-compressed file as mentioned above. I hope this helps anyone else who runs into this issue!
